src/ralph/backends/database/mongo.py

```diff
@@ -0,0 +1,135 @@
+from ralph.exceptions import BadFormatException
+
+from .base import BaseDatabase
+
+logger = logging.getLogger(__name__)
+
+
+class MongoDatabase(BaseDatabase):
+    """Mongo database backend."""
+
+    name = "mongo"
+
+    def __init__(
+        self,
+        connection_uri: str = MONGO_URI,
+        database: str = MONGO_DATABASE,
+        collection: str = MONGO_COLLECTION,
+        client_options: dict = None,
+    ):
+        """Instantiates the Mongo client.
+
+        Args:
+            connection_uri (str): MongoDB connection URI.
+            database (str): MongoDB database to connect to.
+            collection (str): MongoDB database collection to get objects from.
+            client_options (dict): A dictionary of valid options for the MongoClient
+                class initialization.
+        """
+        if client_options is None:
+            client_options = {}
+
+        self.client = MongoClient(connection_uri, **client_options)
+        self.database = getattr(self.client, database)
+        self.collection = getattr(self.database, collection)
+
+    def get(self, chunk_size: int = 500):
+        """Gets collection documents and yields them."""
+
+        for document in self.collection.find(batch_size=chunk_size):
+            # Make the document json-serializable
+            document.update({"_id": str(document.get("_id"))})
+            yield document
+
+    @staticmethod
+    def to_documents(
+        stream: Union[TextIO, list], ignore_errors: bool = False
+    ) -> Generator[dict, None, None]:
+        """Converts `stream` lines (one statement per line) to Mongo documents.
+
+        We expect statements to have at least an `id` field that will be used
+        to compute a unique MongoDB Object ID. This ensures that we will not
+        duplicate statements in our database.
+        """
+
+        for line in stream:
+            statement = json.loads(line) if isinstance(line, str) else line
+            if "id" not in statement:
+                msg = f"statement {statement} has no 'id' field"
+                if ignore_errors:
+                    logger.warning(msg)
+                    continue
+                raise BadFormatException(msg)
+            document = {
+                "_id": ObjectId(
+                    hashlib.sha256(bytes(statement["id"], "utf-8")).hexdigest()[:24]
+                ),
+                "_source": statement,
+            }
+
+            yield document
+
+    def bulk_import(self, batch: list, ignore_errors: bool = False):
+        """Inserts a batch of documents into the selected database collection."""
+
+        try:
+            new_documents = self.collection.insert_many(batch)
+        except BulkWriteError as error:
+            if not ignore_errors:
+                raise error
+            logger.warning(
+                "Bulk importation failed for current documents chunk but you choose "
+                "to ignore it.",
+            )
+            return 0
```

Comment:
```suggestion
            return error.details["nInserted"]
```
It seems pymongo inserts any documents prior to the first failure and omits all subsequent insertions.

Reply:
Apropos, could it be helpful to include the number of inserted/skipped documents in the warning log message? WDYT?